### PR TITLE
Components, Enums, Vendor and Main.go

### PR DESCRIPTION
**Pull request by Dkrenon**
* adds tests and refactors enum -> str and its opposite

### CODE REVIEW

File: cardinal/component/component_action.go [hunk 1]
```diff
@@ -15,6 +15,38 @@ type Action struct {
 	AffectedByActionID uint32           `json:"affected_by_action_id"`
 }
 
+type ActionStore struct {
+	actions map[uint32]Action
+	nextID  uint32
+}
+
+func (ActionStore) Name() string {
```
Comment (lbdl) [inline cardinal/component/component_action.go:23]: do we really need this? as in we can get this from a type assertion, something like if v, ok := interface{}(instance).(MyStruct); ok

File: cardinal/component/component_action.go [hunk 2]
```diff
@@ -15,6 +15,38 @@ type Action struct {
...
+	return "ActionStore"
+}
```
Comment (lbdl) [inline cardinal/component/component_action.go:25]: same as above do we have to return a string for a struct type

File: cardinal/component/component_dirObject.go [hunk 3]
```diff
@@ -5,15 +5,47 @@ import (
...
+
+type DirObjectStore struct {
+	dirObjects map[uint32]DirObject
+	nextID     uint32
 }
 
 func (DirObject) Name() string {
 	return "DirObject"
 }
+
```
Comment (lbdl) [inline cardinal/component/component_dirObject.go:25]: see above on action

File: cardinal/component/component_dirObject.go [hunk 4]
```diff
@@ -5,15 +5,47 @@ import (
...
+func (DirObjectStore) Name() string {
+	return "DirObjectStore"
+}
+
```
Comment (lbdl) [inline cardinal/component/component_dirObject.go:29]: see above on action

File: cardinal/component/component_object.go [hunk 5]
```diff
@@ -5,14 +5,46 @@ import (
...
 
+type ObjectStore struct {
+	objects map[uint32]Object
+	nextID  uint32
+}
+
 func (Object) Name() string {
 	return "Object"
 }
+
```
Comment (lbdl) [inline cardinal/component/component_object.go:24]: same we might be able to use type assertions

File: cardinal/component/component_object.go [hunk 6]
```diff
@@ -5,14 +5,46 @@ import (
...
+func (ObjectStore) Name() string {
+	return "ObjectStore"
+}
+
```
Comment (lbdl) [inline cardinal/component/component_object.go:28]: type assertions

File: cardinal/component/component_room.go [hunk 7]
```diff
@@ -2,19 +2,64 @@ package component
...
+
+type RoomStore struct {
+	rooms  map[uint32]Room
+	nextID uint32
 }
 
 func (Room) Name() string {
 	return "Room"
 }
+
```
Comment (lbdl) [inline cardinal/component/component_room.go:23]: type assertion

File: cardinal/component/component_room.go [hunk 8]
```diff
@@ -2,19 +2,64 @@ package component
...
+func (RoomStore) Name() string {
+	return "RoomStore"
```
Comment (lbdl) [inline cardinal/component/component_room.go:25]: type assertion

File: cardinal/component/component_txtDef.go [hunk 9]
```diff
@@ -5,12 +5,34 @@ import (
...
+	Description string           `json:"description"`
+}
+
+type TxtDefStore struct {
+	txtDefs map[string]TxtDef
 }
 
 func (TxtDef) Name() string {
-	return "TxtDef"
+	return "RTxtDefoom"
```
Comment (lbdl) [inline cardinal/component/component_txtDef.go:18]: type assertion

File: cardinal/component/component_txtDef.go [hunk 10]
```diff
@@ -5,12 +5,34 @@ import (
...
+}
+
+func (TxtDefStore) Name() string {
+	return "TxtDefStore"
```
Comment (lbdl) [inline cardinal/component/component_txtDef.go:22]: type assertion

File: cardinal/component/component_verbData.go [hunk 11]
```diff
@@ -12,6 +12,7 @@ type VerbData struct {
 	Verb            enums.ActionType    // The action/verb from the command
 	DirectNoun      enums.ObjectType    // The direct object of the action
 	IndirectDirNoun enums.DirObjectType // The indirect directional object of the action
+	ObjType         string              // Type of the direct object
```
Comment (lbdl) [inline cardinal/component/component_verbData.go:15]: do we need this? like how will we later use it?

File: cardinal/enums/enums.go [hunk 12]
```diff
@@ -14,10 +14,37 @@ const (
 	RoomTypeFort                       // 5
 	RoomTypeRoom                       // 6
 	RoomTypePlain                      // 7
+	RoomTypeForge                      // 8
 )
 
 func (r RoomType) String() string {
-	return [...]string{"None", "WoodCabin", "Store", "Cavern", "StoneCabin", "Fort", "Room", "Plain"}[r]
+	return [...]string{"None", "WoodCabin", "Store", "Cavern", "StoneCabin", "Fort", "Room", "Plain", "Forge"}[r]
```
Comment (lbdl) [inline cardinal/enums/enums.go]: this should probably use the toStringRoom map defined below

File: cardinal/enums/enums.go [hunk 13]
```diff
@@ -56,6 +109,28 @@ func (d DirObjectType) String() string {
 	return [...]string{"None", "Door", "Window", "Stairs", "Ladder", "Path", "Trail"}[d]
```
Comment (lbdl) [inline cardinal/enums/enums.go]: should probably use the toStringDObject map

File: cardinal/enums/enums.go [hunk 14]
```diff
@@ -107,6 +232,28 @@ func (o ObjectType) String() string {
 	return [...]string{"None", "Football", "Key", "Knife", "Bottle", "Straw", "Petrol"}[o]
```
Comment (lbdl) [inline cardinal/enums/enums.go]: again shouls likely use the map instead

File: cardinal/enums/enums.go [hunk 15]
```diff
@@ -142,3 +317,23 @@ const (
 func (t TxtDefType) String() string {
 	return [...]string{"None", "DirObject", "Dir", "Place", "Object", "Action"}[t]
```
Comment (lbdl) [inline cardinal/enums/enums.go]: same as above use the map

File: cardinal/vendor/github.com/davecgh/go-spew/LICENSE
Comment (lbdl) [inline cardinal/vendor/github.com/davecgh/go-spew/LICENSE:1]: these files are mainly 3rd party and as such Im not sure that these should be checked in...

**Review by lbdl — COMMENTED** [review]
good work lets work through the comments

File: cardinal/component/component_action.go (hunk 1 above)
Comment (Dkrenon) [reply cardinal/component/component_action.go:23]: The thing is that is how the documentation from WE regarding cardinal components has. If that is missing, we will get the following problem
`[{"message": "\"github.com/ArchetypalTech/TheOrugginTrail-ArgusWE/cardinal/component\".Action does not satisfy \"pkg.world.dev/world-engine/cardinal/types\".Component (missing method Name)",}]`
Therefore, it is needed for all components

File: cardinal/component/component_action.go (hunk 2 above)
Comment (Dkrenon) [reply cardinal/component/component_action.go:25]: same as above

File: cardinal/component/component_dirObject.go (hunk 3 above)
Comment (Dkrenon) [reply cardinal/component/component_dirObject.go:25]: same as first one

File: cardinal/component/component_dirObject.go (hunk 4 above)
Comment (Dkrenon) [reply cardinal/component/component_dirObject.go:29]: same as first one

File: cardinal/component/component_object.go (hunk 5 above)
Comment (Dkrenon) [reply cardinal/component/component_object.go:24]: same as first one

File: cardinal/component/component_object.go (hunk 6 above)
Comment (Dkrenon) [reply cardinal/component/component_object.go:28]: same as first one

File: cardinal/component/component_room.go (hunk 7 above)
Comment (Dkrenon) [reply cardinal/component/component_room.go:23]: same as first one

File: cardinal/component/component_room.go (hunk 8 above)
Comment (Dkrenon) [reply cardinal/component/component_room.go:25]: same as first one

File: cardinal/component/component_txtDef.go (hunk 9 above)
Comment (Dkrenon) [reply cardinal/component/component_txtDef.go:18]: same as first one

File: cardinal/component/component_txtDef.go (hunk 10 above)
Comment (Dkrenon) [reply cardinal/component/component_txtDef.go:22]: same as first one

**Review by lbdl — APPROVED** [review]

**Pull request merged**